docs/proposals/helm-manifest-deployer-ordering.md

```diff
@@ -0,0 +1,251 @@
+
+- Namespaced objects deployed by the Chart
+- Not namespaced objects deployed by the Chart
+- CRDs deployed by the Chart
+
+The algorithm does not wait until particular objects are gone before it continues deleting the next ones.The removal of 
+a Chart is successful if all objects were gone. Objects not deployed by the Chart, e.g. custom resources (CRs) deployed
+by some operator/job are not removed.
+
+### New Solution
+
+## Default Deletion Behaviour
+
+We propose the following solution to have more control over the deletion process.
+
+The basic order of deleting the deployed manifests remains the same as before and is divided into three deletion groups:
+
+- Namespaced objects deployed by the Chart
+- Not namespaced objects deployed by the Chart (except CRDs)
+- CRDs deployed by the Chart
+
+The deletion continues only with the next object/deletion group if all objects from the groups before are gone. This is 
+different to the current approach. The deletion is tried as long until all objects of all 3 deletion groups are gone
+or the specified configurable timeout (default 5 min.) comes into the game and the deletion failed.
+
+## Custom Deletion Behaviour
+
+### Basics
+
+To change the deletion behaviour of a DeployItem, you could specify your own custom deletion groups. A deletion group describes
+a set of k8s objects which should be deleted. The deletion groups are defined as a list and one deletion group after 
+the other is processed. Thereby again, all objects of a deletion group must be gone before the deletion of objects of the next 
+deletion group starts. This is tried until all objects of all deletion groups are gone (SUCCESS) or the timeout comes 
+into place (FAILED). If you have specified your own deletion groups, the default deletion behaviour is completely disabled.
+
+Custom deletion groups are defined in the section `deletionGroups` as shown below:
+
+```yaml
+deployItems:
+  - name: my-deploy-item
+    type: landscaper.gardener.cloud/helm
+    config:
+      apiVersion: helm.deployer.landscaper.gardener.cloud/v1alpha1
+      kind: ProviderConfiguration
+      ...
+      deletionGroups:
+      - <deletionGroup-1>
+      - <deletionGroup-2>
+      ...
+```
+
+### Predefined Resources
+
+The most simple specification for a deletion group is using predefined resource groups. The following example
+shows how to specify the default deletion behaviour with 3 such deletion groups:
+
+
+```yaml
+deployItems:
+  - name: my-deploy-item
+    type: landscaper.gardener.cloud/helm
+    config:
+      apiVersion: helm.deployer.landscaper.gardener.cloud/v1alpha1
+      kind: ProviderConfiguration
+      ...
+      deletionGroups:
+        - predefinedResourceGroup: 
+            type: namespaced-resources
+        - predefinedResourceGroup:
+            type: cluster-scoped-resources     # does not include the crds
+        - predefinedResourceGroup: 
+            type: crds
```

Comment:
```suggestion
      deletionGroups:
        predefinedResourceGroups: 
          - type: namespaced-resources
          - type: cluster-scoped-resources     # does not include the crds
          - type: crds
```

Wouldn't this beaser to write and parse (as a human :-) )?

Reply:
resolved as discussed offline